libafl/src/inputs/mod.rs

```diff
@@ -198,36 +197,44 @@ pub trait HasMutatorBytes: HasLen {
     }
 }
 
-/// Mapping types to themselves, used to ensure lifetime consistency for mapped mutators.
-///
-/// Specifically, this is for [`Input`] types that are owned wrappers around a reference. The lifetime of the associated type should be the same as the reference.
-pub trait MappedInput {
-    /// The type for which this trait is implemented
-    type Type<'a>
+impl HasMutatorBytes for Vec<u8> {
+    fn bytes(&self) -> &[u8] {
+        self.as_ref()
+    }
+
+    fn bytes_mut(&mut self) -> &mut [u8] {
+        self.as_mut()
+    }
+
+    fn resize(&mut self, new_len: usize, value: u8) {
```

Comment:
Doesn't this make using `Vec` a bit of a headache whenever `HasMutatorBytes` is in scope?

Reply:
If, in the same file, one has `HasMutatorBytes` in scope and uses any of the duplicate functions on a `Vec<u8>` specifically, one would need to specify which implementation to use. Since one just forwards to the other, you can't really choose wrong, it's just annoying to write the extra code.